### PR TITLE
Feature/#31  update

### PR DESCRIPTION
**Pull request by dankelleher**
Adds update features to the client and driver - including the choice of whether to append or overwrite (default is append)

### CODE REVIEW

**Comment by dankelleher** [comment]
Merge https://github.com/identity-com/solid-did/pull/32 first

File: client/src/lib/solana/solid-data.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { clusterApiUrl, Cluster, PublicKey } from '@solana/web3.js';
 import { Assignable, Enum, SCHEMA } from './solana-borsh';
 import { DID_METHOD, DID_HEADER } from '../constants';
 import { encode } from 'bs58';
+import { mergeWith } from 'ramda';
```
Comment (joncinque) [inline client/src/lib/solana/solid-data.ts:5]: I didn't know about this library before, really neat!

File: client/src/lib/solana/transaction.ts [hunk 2]
```diff
@@ -38,4 +45,45 @@ export class SolidTransaction {
...
+      closeAccount(recordKey, payer.publicKey, payer.publicKey)
+    );
+
+    // Send the instructions
+    return SolanaUtil.sendAndConfirmTransaction(connection, transaction, payer);
+  }
+
+  static async updateSolid(
+    connection: Connection,
+    payer: Account,
```
Comment (joncinque) [inline client/src/lib/solana/transaction.ts:67]: We can also add the authority separately for this instruction

File: client/test/unit/lib/solana/solid-data.test.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  ServiceEndpoint,
+  SolidData,
+} from '../../../../src/lib/solana/solid-data';
+import { Account } from '@solana/web3.js';
+import { omit } from 'ramda';
+import { makeService } from '../../../util';
+
+const pub = () => new Account().publicKey;
+
+describe('solid-data', () => {
```
Comment (joncinque) [inline client/test/unit/lib/solana/solid-data.test.ts:12]: This is really great too

**Review by joncinque — COMMENTED** [review]
Looks good, just needs to separate the payer from the authority

File: client/src/lib/solana/solid-data.ts (hunk 1 above)
Comment (dankelleher) [reply client/src/lib/solana/solid-data.ts:5]: probably my favourite JS lib :)

File: client/src/lib/solana/transaction.ts (hunk 2 above)
Comment (dankelleher) [reply client/src/lib/solana/transaction.ts:67]: Agreed. I was thinking along the lines of keeping the API simple but you are probably right

**Review by joncinque — APPROVED** [review]
Just the one nit to think about it, looks good!

**Pull request merged**